cosmwasm/its.ts

```diff
@@ -67,19 +67,88 @@ export async function checkSingleTokenRegistration(
-    const supply = await token.totalSupply();
-    return supply.toString();
+export async function modifyTokenSupply(
+    client: ClientManager,
+    config: ConfigManager,
+    interchainTokenServiceAddress: string,
+    tokenId: string,
+    tokenAddress: string,
+    chain: string,
+    dryRun: boolean,
+) {
+    const tokenRegistered = await checkSingleTokenRegistration(config, client, interchainTokenServiceAddress, tokenId, chain);
+    if (!tokenRegistered) {
+        printInfo(`Token ${tokenId} on ${chain} is not registered`);
+        return;
+    }
+
+    const { origin_chain } = await client.queryContractSmart(interchainTokenServiceAddress, {
+        token_config: { token_id: formatTokenId(tokenId) },
+    });
+
+    if (origin_chain === config.getChainConfig(chain).axelarId) {
+        printInfo(`Token ${tokenId} origin chain is ${chain}, it should be set to untracked.`);
+        return;
+    }
+
+    const { supply, isTokenSupplyTracked } = await getTokenInstanceInfo(tokenAddress, config.getChainConfig(chain).rpc);
+
+    if (!isTokenSupplyTracked) {
+        printInfo(`Token ${tokenId} on ${chain} supply should not be tracked`);
+        return;
+    }
+
+    const tokenInstanceOnHub = await client.queryContractSmart(interchainTokenServiceAddress, {
+        token_instance: { chain: config.getChainConfig(chain).axelarId, token_id: formatTokenId(tokenId) },
+    });
+
+    let supplyOnHub: bigint;
+    if (tokenInstanceOnHub.supply === 'untracked') {
+        supplyOnHub = BigInt(0);
+    } else {
+        supplyOnHub = BigInt(tokenInstanceOnHub.supply.tracked);
+    }
+
+    if (supply === supplyOnHub) {
+        printInfo(`Token ${tokenId} on ${chain} supply is up-to-date`);
+        return;
+    }
+
+    const supplyModifier = supply > supplyOnHub ? 'increase_supply' : 'decrease_supply';
+    const supplyDifference = supply > supplyOnHub ? supply - supplyOnHub : supplyOnHub - supply;
+
+    const msg = {
+        modify_supply: {
+            chain: config.getChainConfig(chain).axelarId,
+            token_id: formatTokenId(tokenId),
+            supply_modifier: {
+                [supplyModifier]: supplyDifference.toString(),
+            },
+        },
+    };
+
+    const [account] = client.accounts;
+
+    if (!dryRun) {
+        await client.execute(account.address, interchainTokenServiceAddress, msg, 'auto');
+    }
 }
 
-// TODO tkulik: This command will be used in the supply alignment command.
 export async function isTokenSupplyTracked(tokenManagerType: number, token: Contract): Promise<boolean> {
     return tokenManagerType === tokenManagerTypes.NATIVE_INTERCHAIN_TOKEN && (await token.isMinter(constants.AddressZero));
 }
 
+export async function getTokenInstanceInfo(tokenAddress: string, rpc: string): Promise<{ supply: bigint; isTokenSupplyTracked: boolean }> {
+    const provider = getDefaultProvider(rpc);
+    const token = new Contract(tokenAddress, IInterchainToken.abi, provider);
```

Comment:
It's not guaranteed that the token is an interchain token. We can modify supply for any token registered on ITS (e.g. custom tokens like XRP that aren't deployed by ITS)